ingest/src/test/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClientTest.java

```diff
@@ -263,24 +306,35 @@ void IngestFromFile_CompressedJson() throws Exception {
         }
         assertEquals(OperationStatus.Succeeded, status);
         assertTrue(visited.get());
+        verifyClientRequestId();
     }
 
-    @Test
-    void IngestFromStream_Success() throws Exception {
+
+    @ParameterizedTest
+    @CsvSource({"true,true", "false,true", "true,false", "false,false"})
+    void IngestFromStream_Success(boolean leaveOpen, boolean useSourceId) throws Exception {
         String data = "Name, Age, Weight, Height";
-        InputStream inputStream = new ByteArrayInputStream(StandardCharsets.UTF_8.encode(data).array());
-        StreamSourceInfo streamSourceInfo = new StreamSourceInfo(inputStream);
+        InputStream inputStream = new CloseableByteArrayInputStream(StandardCharsets.UTF_8.encode(data).array());
+        UUID sourceId = useSourceId ? CustomUUID : null;
+        StreamSourceInfo streamSourceInfo = new StreamSourceInfo(inputStream, leaveOpen, sourceId);
         OperationStatus status = managedStreamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
         assertEquals(OperationStatus.Succeeded, status);
         verify(streamingClientMock, atLeastOnce()).executeStreamingIngest(any(String.class), any(String.class), argumentCaptor.capture(),
-                isNull(), any(String.class), eq("mappingName"), any(boolean.class));
+                clientRequestPropertiesCaptor.capture(), any(String.class), eq("mappingName"), any(boolean.class));
 
         /* In order to make efficient ingestion requests, the streaming ingest client compress the given stream unless it is already compressed.
          * When the given stream content is already compressed, the user must specify that in the stream source info.
          * This method verifies if the stream was compressed correctly.
          */
         InputStream stream = argumentCaptor.getValue();
         verifyCompressedStreamContent(stream, data);
+        verifyClientRequestId(0, sourceId);
+        if (leaveOpen) {
+            assertDoesNotThrow(() -> inputStream.read(new byte[1]));
+        } else {
+            //noinspection ResultOfMethodCallIgnored - we are expecting an exception, so we don't care about the result
```

Comment:
I'm not a big fan of adding these, especially if the only issue we have without them is a SonarLint warning. Personally, I'd rather make a dummy assignment if we're bothered by the warning.

Reply:
Replaced with assignment, looks good?